tpot/driver.py

```diff
@@ -65,176 +65,406 @@ def float_range(value):
+        help='Flag indicating whether the TPOT version checker should be disabled.'
+    )
+
+    parser.add_argument(
+        '--version',
+        action='version',
         version='TPOT {version}'.format(version=__version__),
-        help='Show the TPOT version number and exit.')
+        help='Show the TPOT version number and exit.'
+    )
+
+    return parser
+
+
+def _print_args(args):
+    print('\nTPOT settings:')
+    args.__dict__
+    for arg, arg_val in list(enumerate(args.__dict__)):
+        if arg == 'DISABLE_UPDATE_CHECK':
+            continue
+        elif arg == 'SCORING_FN' and arg_val is None:
+            if args.TPOT_MODE == 'classification':
+                arg_val = 'accuracy'
+            else:
+                arg_val = 'mean_squared_error'
+        elif arg == 'OFFSPRING_SIZE' and arg_val is None:
+            arg_val = args.__dict__['POPULATION_SIZE']
+        print('{}\t=\t{}'.format(arg, arg_val))
+    print('')
 
-    args = parser.parse_args()
 
-    if args.VERBOSITY >= 2:
-        print('\nTPOT settings:')
-        for arg in sorted(args.__dict__):
-            arg_val = args.__dict__[arg]
-            if arg == 'DISABLE_UPDATE_CHECK':
-                continue
-            elif arg == 'SCORING_FN' and arg_val is None:
-                if args.TPOT_MODE == 'classification':
-                    arg_val = 'accuracy'
-                else:
-                    arg_val = 'mean_squared_error'
-            elif arg == 'OFFSPRING_SIZE' and arg_val is None:
-                arg_val = args.__dict__['POPULATION_SIZE']
-            print('{}\t=\t{}'.format(arg, arg_val))
-        print('')
-
-    input_data = np.recfromcsv(args.INPUT_FILE, delimiter=args.INPUT_SEPARATOR, dtype=np.float64, case_sensitive=True)
+def _read_data_file(args):
+    input_data = np.recfromcsv(
+        args.INPUT_FILE,
+        delimiter=args.INPUT_SEPARATOR,
+        dtype=np.float64,
+        case_sensitive=True
+    )
+
     if args.TARGET_NAME not in input_data.dtype.names:
-        raise ValueError('The provided data file does not seem to have a target column. '
-                         'Please make sure to specify the target column using the -target parameter.')
+        raise ValueError(
+            'The provided data file does not seem to have a target column. '
+            'Please make sure to specify the target column using the -target '
+            'parameter.'
+        )
 
-    features = np.delete(input_data.view(np.float64).reshape(input_data.size, -1),
-                         input_data.dtype.names.index(args.TARGET_NAME), axis=1)
+    return input_data
 
-    training_features, testing_features, training_classes, testing_classes = \
-        train_test_split(features, input_data[args.TARGET_NAME], random_state=args.RANDOM_STATE)
 
-    if args.TPOT_MODE == 'classification':
-        tpot_type = TPOTClassifier
-    else:
-        tpot_type = TPOTRegressor
+def _impute_missing_values(features, missing_value):
```

Comment:
I suggest that this imputation function could also be added into interactive mode by moving it to base.py